ludwig/schema/utils.py

```diff
@@ -63,6 +65,40 @@ def load_config_with_kwargs(
     }
 
 
+def convert_submodules(config_dict):
+    """Helper function for converting submodules to dictionaries during a config object to dict transformation.
+
+    Args:
+        config_dict: Top level config dictionary with un-converted submodules
+
+    Returns:
+        The fully converted config dictionary
+    """
+    output_dict = copy.deepcopy(config_dict)
+
+    for k, v in output_dict.items():
```

Comment:
This is really cool. Though it seems like if I were to implement a `__deepcopy__()` on the the `BaseMarshmallowConfig` we would still need to call `deepcopy()`. I feel like I may be misunderstanding the need for this but at least my interpretation of this was if there is some specific custom deep copy logic that we may need to implement, but it seems like the regular deep copy functionality works. WDYT?

Reply:
The way I read is that you wouldn’t then need to enumerate the child properties like you are doing, but happy to take you approach here as I don’t have a lot of context.

Reply:
Funny enough as I look through it, I don't think I even need the elif section where I enumerate, so going to remove that. I do think that we still need a deep copy though because there are nested defaults that we don't want to have loose pointers assigned to them. Will make sure to double check this though to see if we can remove that iff possible.